src/site_store.rs

```diff
@@ -1,89 +1,91 @@
-                !name.ends_with(extension) && name.starts_with(&format!("{basename}."))
-            });
-            if remove {
-                remove_file(file.path())?;
-                removed_something = true;
+    let mut superfluous = Vec::new();
+    for entry in read_dir(".")? {
+        let entry = entry?.path();
+        if !entry.is_dir() {
+            continue;
+        }
+
+        if let Some(filename) = entry.file_name() {
+            let is_relevant = filename.to_str().map_or(false, |o| !o.starts_with('.'));
+            if is_relevant {
+                superfluous.append(&mut inner(expected_paths, Path::new(filename))?);
             }
         }
-        Ok(removed_something)
     }
+    superfluous.sort();
+    Ok(superfluous)
+}
 
-    pub fn write_only_changed(
-        &self,
-        basename: &str,
-        extension: &str,
-        content: &str,
-    ) -> anyhow::Result<ChangeKind> {
-        let path = format!("{}/{basename}.{extension}", self.folder);
-        let content = content.trim().to_string() + "\n";
+fn remove_same_base_different_extension(path: &Path) -> anyhow::Result<bool> {
+    let parent = path.parent().unwrap_or_else(|| Path::new("."));
+    let stem = path.file_stem();
+    let extension = path.extension();
+    let mut removed_something = false;
+    for file in read_dir(parent)? {
+        let file = file?.path();
 
-        let current = read_to_string(&path).unwrap_or_default();
-        let changed = current != content;
-        if changed {
-            write(&path, content)?;
+        let same_stem = file.file_stem() == stem;
+        if !same_stem {
+            continue;
         }
 
-        let removed_something = self.remove_same_base_different_extension(basename, extension)?;
+        let same_extension = file.extension() == extension;
+        let remove = !same_extension;
 
-        if removed_something {
-            Ok(ChangeKind::Changed)
-        } else if current.is_empty() {
-            Ok(ChangeKind::Init)
-        } else if changed {
-            Ok(ChangeKind::Changed)
-        } else {
-            Ok(ChangeKind::ContentSame)
+        if remove {
+            remove_file(file)?;
+            removed_something = true;
         }
     }
+    Ok(removed_something)
 }
 
-fn basename_is_wanted(basenames: &[String], searched: &str) -> bool {
-    for basename in basenames {
-        if searched.starts_with(&format!("{basename}.")) {
-            return true;
-        }
+pub fn write_only_changed(path: &Path, content: &str) -> anyhow::Result<ChangeKind> {
+    if let Some(parent) = path.parent() {
+        create_dir_all(parent)?;
+    }
+    let content = content.trim().to_string() + "\n";
+
+    let current = read_to_string(path).unwrap_or_default();
+    let changed = current != content;
+    if changed {
+        write(path, content)?;
+    }
+
+    let removed_something = remove_same_base_different_extension(path)?;
```

Comment:
Why?

Reply:
When the config changes and the resulting file is a `md` instead of a `html` now the old one with a different extension is cleaned with this.

Reply:
Maybe add that as a comment? 👀